packages/page-claims/src/Attest.tsx

```diff
@@ -2,33 +2,70 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import React from 'react';
+import React, { useEffect, useCallback, useState } from 'react';
 import styled from 'styled-components';
 import { Button, Card, TxButton } from '@polkadot/react-components';
+import { FormatBalance } from '@polkadot/react-query';
+import { Option } from '@polkadot/types';
+import { BalanceOf, EthereumAddress, StatementKind } from '@polkadot/types/interfaces';
 
 import { useTranslation } from './translate';
-import { addrToChecksum } from './util';
+import { useApi } from '@polkadot/react-hooks';
 
 interface Props {
   accountId: string;
   className?: string;
+  ethereumAddress: EthereumAddress | null;
+  statementKind?: StatementKind;
 }
 
-function Attest ({ accountId, className }: Props): React.ReactElement<Props> | null {
+function Attest ({ accountId, className, ethereumAddress, statementKind }: Props): React.ReactElement<Props> | null {
   const { t } = useTranslation();
+  const { api } = useApi();
+  const [claimValue, setClaimValue] = useState<BalanceOf | null>(null);
+  const [claimAddress, setClaimAddress] = useState<EthereumAddress | null>(null);
+  const [isBusy, setIsBusy] = useState(false);
+
+  const _fetchClaim = useCallback(
+    (address: EthereumAddress): void => {
+      setIsBusy(true);
+
+      api.query.claims
+        .claims<Option<BalanceOf>>(address)
+        .then((claim): void => {
+          setClaimValue(claim.unwrapOr(null));
+          setIsBusy(false);
+        })
+        .catch((): void => setIsBusy(false));
+    },
+    [api]
+  );
+
+  useEffect((): void => {
+    setClaimAddress(ethereumAddress);
+
+    ethereumAddress && _fetchClaim(ethereumAddress);
+  }, [_fetchClaim, ethereumAddress]);
+
+  if (isBusy || !claimAddress) {
+    return null;
+  }
+
+  const hasClaim = claimValue && claimValue.gten(0);
 
   return (
     <Card isSuccess>
       <div className={className}>
         {t('By signing this transaction, you agree with the attestation available at the following address: XXX')}
-        <h3>{addrToChecksum(accountId)}</h3>
+        {hasClaim &&
+          <h3>{t('Attestation for: ')} <FormatBalance value={claimValue} /></h3>}
         <Button.Group>
           <TxButton
             accountId={accountId}
             icon='send'
             isPrimary
             label={t('Attest')}
-            params={['dsafdsalkfj']} // Replace with actual
+            params={[statementKind]} // Replace with actual
```

Comment:
iiuc, we shouldn't attest the statementKind, but the actual statement in english: https://github.com/paritytech/polkadot/pull/1091/files#diff-6713ee0e82ee03f026aa21e5e99fb055R66

i.e. `statementKind` is an Enum, but we should put here the string matching the enum

Reply:
We can leave as-is for now, as anything will work, we it should probably be replaced at some point with hardcoded strings like:
```tsx
statementKind.siDefault() ? 'statement1' : 'statement2'
```

Reply:
yup absolutely, that's my understanding as well. Although right now the `statement === statemendKind`